indexer/README.md

```diff
@@ -1,2 +1,80 @@
+
+-**JSON parser**:
+        ```
+        def json_parser(json_file_path: str):
+            with open(json_file_path, "r") as file:
+                data = json.load(file)
+            return json.dumps(data, indent=4)```
+
+    The parser function json_parser() is called when the file to be parsed is of json format. This function uses the built-in package JSON, which can be used to work with JSON data. This function first opens the file provided (json_file_path) in read only mode using `open(json_file_path, "r")`. The file is parsed using `json.load()` method which takes a file object and returns a json object. Here, `data = json.load(file)` parses the file and gives us a dictionary named data, that contains the contents of the file. `json.dumps(data, indent=4)` takes the "data" object, converts it back into a JSON-formatted string, and applies pretty-printing with an indentation level of 4 spaces.
+
+-**Default parser**:
+        ```
+        def default_parser(file_path: str):
+            with open(file_path, "r") as file:
+                return file.read()```
+    
+    When the file to be parsed is not of pdf, xslx, json or docx format, then the default_parser() function is called to parse the file. This function opens the file provided (file_path) in "r" mode i.e. read only mode and returns the contents of the file as a string.
+
+
+## Chunking strategy and process:
+
+Once the files are parsed, the indexer service breaks down the collected information and arranges it into neat, searchable chunks. This helps in finding the right information quickly. The indexer breaks down the knowledge fed into Jugalbandi into smaller parts and adds labels to each part, making it easy to search and relate to other information.
+
+### Chunking Parameters
+
+The chunking process uses the following parameters:
+
+- `chunk_size`: Typically **4000 characters**
+- `chunk_overlap`: Usually **200 characters**
+- List of separators (e.g., newline, period)
+
+These parameters ensure smooth, logical breaks in the content.
+
+- The `chunk_size` value indicates the maximum number of characters allowed in each chunk.
+- The `chunk_overlap` value indicates how much overlap of characters must be present between consecutive chunks.
+
+### Chunking Process
+
+The content collected from each file is chunked in such a way that:
+
+- Every chunk has a maximum of 4000 characters.
+- The first **200 characters overlap** with the last 200 characters of the previous chunk.
+- No sentences are cut in between chunks.
+
+This results in continuous and logically structured chunks for efficient searching and retrieval.
+
+
+## API and Indexer Service Workflow
```

Comment:
Remove everything in these ## API and Indexer Service Workflow I will add it

Reply:
Done